pkg/apis/core/validation/secretbinding.go

```diff
@@ -42,6 +45,9 @@ func ValidateSecretBindingUpdate(newBinding, oldBinding *core.SecretBinding) fie
 	allErrs = append(allErrs, apivalidation.ValidateObjectMetaUpdate(&newBinding.ObjectMeta, &oldBinding.ObjectMeta, field.NewPath("metadata"))...)
 	allErrs = append(allErrs, apivalidation.ValidateImmutableField(newBinding.SecretRef, oldBinding.SecretRef, field.NewPath("secretRef"))...)
 	allErrs = append(allErrs, apivalidation.ValidateImmutableField(newBinding.Quotas, oldBinding.Quotas, field.NewPath("quotas"))...)
+	if utilfeature.DefaultFeatureGate.Enabled(features.ImmutableSecretBindingProvider) {
```

Comment:
To be able to combine the 2 feature gates `RequiredSecretBindingProvider` and `ImmutableSecretBindingProvider` the following additional check would be required here:
```suggestion
	if utilfeature.DefaultFeatureGate.Enabled(features.ImmutableSecretBindingProvider)  && oldBinding.Provider != nil {
```